be1-go/channel/lao/mod.go

```diff
@@ -0,0 +1,738 @@
+
+	// Saving the election channel creation message on the lao channel
+	c.inbox.StoreMessage(msg)
+
+	// Add the new election channel to the organizerHub
+	c.hub.RegisterNewChannel(channelPath, &electionCh)
+
+	return nil
+}
+
+// processCreateRollCall processes a roll call creation object.
+func (c *Channel) processCreateRollCall(msg messagedata.RollCallCreate) error {
+	// Check that the ProposedEnd is greater than the ProposedStart
+	if msg.ProposedStart > msg.ProposedEnd {
+		return answer.NewErrorf(-4, "The field `proposed_start` is greater than the field `proposed_end`: %d > %d", msg.ProposedStart, msg.ProposedEnd)
+	}
+
+	c.rollCall.id = string(msg.ID)
+	c.rollCall.state = Created
+	return nil
+}
+
+// processOpenRollCall processes an open roll call object.
+func (c *Channel) processOpenRollCall(msg message.Message, action string) error {
+	if action == "open" {
+		// If the action is an OpenRollCallAction,
+		// the previous roll call action should be a CreateRollCallAction
+		if c.rollCall.state != Created {
+			return answer.NewError(-1, "The roll call cannot be opened since it does not exist")
+		}
+	} else {
+		// If the action is an RepenRollCallAction,
+		// the previous roll call action should be a CloseRollCallAction
+		if c.rollCall.state != Closed {
+			return answer.NewError(-1, "The roll call cannot be reopened since it has not been closed")
+		}
+	}
+
+	// Why not messagedata.RollCallReopen ? Maybe we should assume that Reopen
+	// message is useless.
+	var rollCallOpen messagedata.RollCallOpen
+
+	err := msg.UnmarshalData(&rollCallOpen)
+	if err != nil {
+		return xerrors.Errorf("failed to unmarshal roll call open: %v", err)
+	}
+
+	if !c.rollCall.checkPrevID([]byte(rollCallOpen.Opens)) {
+		return answer.NewError(-1, "The field `opens` does not correspond to the id of the previous roll call message")
+	}
+
+	c.rollCall.id = string(rollCallOpen.UpdateID)
+	c.rollCall.state = Open
+	return nil
+}
+
+// processCloseRollCall processes a close roll call message.
+func (c *Channel) processCloseRollCall(msg messagedata.RollCallClose) error {
+	if c.rollCall.state != Open {
+		return answer.NewError(-1, "The roll call cannot be closed since it's not open")
+	}
+
+	if !c.rollCall.checkPrevID([]byte(msg.Closes)) {
+		return answer.NewError(-4, "The field `closes` does not correspond to the id of the previous roll call message")
+	}
+
+	c.rollCall.id = msg.UpdateID
+	c.rollCall.state = Closed
+
+	var db *sql.DB
+
+	if os.Getenv("HUB_DB") != "" {
+		db, err := sql.Open("sqlite3", os.Getenv("HUB_DB"))
+		if err != nil {
+			log.Printf("error: failed to connect to db: %v", err)
+			db = nil
+		} else {
+			defer db.Close()
+		}
+	}
+
+	for _, attendee := range msg.Attendees {
+		c.attendees[attendee] = struct{}{}
+
+		if db != nil {
+			log.Printf("inserting attendee into db")
+
+			err := insertAttendee(db, attendee, c.channelID)
+			if err != nil {
+				log.Printf("error: failed to insert attendee into db: %v", err)
+			}
+		}
+	}
```

Comment:
same thing, I'd love (some) of it to just be embedded within the Inbox and hidden away from the LAO channel.

Reply:
Thing is the attendees are stored on the hub (organizer) and not in the inbox. Thus the need to have db handling in both places.